### PR TITLE
#1-4Task.Alex Malyshko

### PR DESCRIPTION
**Pull request by Hem3X**
https://travis-ci.com/github/Hem3X/js-assignments/builds/176207382

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    return (value.slice(7, value.length - 1));
```
Comment (KostyaKov) [inline task/01-strings-tasks.js]: Here you can use "slice(7, -1)" without extra brackets

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -181,7 +190,9 @@ function getParallelipidedDiagonal(a,b,c) {
  *   1678, 3  => 2000
  */
 function roundToPowerOfTen(num, pow) {
-    throw new Error('Not implemented');
+    num = num / 10 ** pow;
+    num = Math.round(num) * 10 ** pow;
```
Comment (KostyaKov) [inline task/02-numbers-tasks.js]: You can also use Math.pow

File: task/04-arrays-tasks.js [hunk 3]
```diff
@@ -70,7 +77,11 @@ function doubleArray(arr) {
  *    [] => [] 
  */
 function getArrayOfPositives(arr) {
-   throw new Error('Not implemented');
+   let result = [];
+   arr.map(function(elem) {
```
Comment (KostyaKov) [inline task/04-arrays-tasks.js]: Here you can use "filter" function on array

File: task/04-arrays-tasks.js [hunk 4]
```diff
@@ -85,7 +96,11 @@ function getArrayOfPositives(arr) {
  *    [ 'cat, 'dog', 'raccon' ] => [ 'cat', 'dog', 'racoon' ]
  */
 function getArrayOfStrings(arr) {
-   throw new Error('Not implemented');
+   let result = [];
+   arr.map(function(elem) {
```
Comment (KostyaKov) [inline task/04-arrays-tasks.js]: Here you can use "filter" function on array

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -200,7 +219,11 @@ function getTail(arr, n) {
  *    +'30,31,32,33,34'
  */
 function toCsvText(arr) {
-   throw new Error('Not implemented');
+   let result =  arr.map(function(elem, index) {
```
Comment (KostyaKov) [inline task/04-arrays-tasks.js]: For this task you simply can use "join" function on array

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -321,7 +359,19 @@ function getPositivesCount(arr) {
  *   [ 'one','one','one','zero' ]     => [ 'zero','one','one','one' ]
  */
 function sortDigitNamesByNumericOrder(arr) {
-   throw new Error('Not implemented');
+   function nameToDigit(str){
+      if (str.toLowerCase() === 'zero') return 0;
+      else if (str.toLowerCase() === 'one') return 1;
```
Comment (KostyaKov) [inline task/04-arrays-tasks.js]: "else" is redundant after "return"

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by Hem3X** [comment]
https://travis-ci.com/github/Hem3X/js-assignments/builds/181762199

**Review by KostyaKov — COMMENTED** [review]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**